### PR TITLE
remove TestSuite structure

### PR DESCRIPTION
**Pull request by jmccormick2001**
**Description of the change:**

this PR removes the TestSuite golang structure from the scorecard code. This is a general
code refactoring and doesn't impact the end-user output or the documentation.

**Motivation for the change:**

 The TestSuite structure no longer served a purpose the addition of labels onto the tests themselves.

### CODE REVIEW

File: internal/scorecard/plugins/plugin_runner.go [hunk 1]
```diff
@@ -178,59 +179,63 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig,
...
+		result.Suggestions = make([]string, 0)
+		result.Errors = make([]error, 0)
+		testResults = append(testResults, result)
+	}
+
+	output := scapiv1alpha2.NewScorecardOutput()
+	output.Log = ""
+
+	for _, tr := range testResults {
+		output.Results = append(output.Results, testResultToScorecardTestResult(tr))
```
Comment (joelanford) [inline internal/scorecard/plugins/plugin_runner.go]: Nit: It seems like we repeat this code in a few places. Would it make sense to pass the results and log values to `scapiv1alpha2.NewScorecardOutput`?

File: internal/scorecard/plugins/labels_test.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package scplugins
 import (
 	"testing"
 
+	schelpers "github.com/operator-framework/operator-sdk/internal/scorecard/helpers"
```
Comment (estroz) [inline internal/scorecard/plugins/labels_test.go]:
```suggestion
	schelpers "github.com/operator-framework/operator-sdk/internal/scorecard/helpers"

```

File: internal/scorecard/plugins/labels_test.go [hunk 3]
```diff
@@ -51,9 +52,15 @@ func TestBasicShortNames(t *testing.T) {
...
 
-			basicTests := NewBasicTestSuite(BasicTestConfig{})
-			basicTests.ApplySelector(selector)
-			testsSelected := len(basicTests.Tests)
+			tests := make([]schelpers.Test, 0)
+
+			conf := BasicTestConfig{}
+			tests = append(tests, NewCheckSpecTest(conf))
+			tests = append(tests, NewCheckStatusTest(conf))
+			tests = append(tests, NewWritingIntoCRsHasEffectTest(conf))
```
Comment (estroz) [inline internal/scorecard/plugins/labels_test.go]: This code would be idiomatic if each line were combined into one append:
```go
			tests = append(tests,
				NewCheckSpecTest(conf),
				NewCheckStatusTest(conf),
				NewWritingIntoCRsHasEffectTest(conf),
			)
```
Ditto below.

File: internal/scorecard/plugins/plugin_runner.go [hunk 4]
```diff
@@ -178,59 +179,63 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig,
...
+		tests = append(tests, NewCRDsHaveValidationTest(conf))
+		tests = append(tests, NewCRDsHaveResourcesTest(conf))
+		tests = append(tests, NewSpecDescriptorsTest(conf))
+		tests = append(tests, NewStatusDescriptorsTest(conf))
+	}
+
+	tests = applySelector(tests, config.Selector)
+
+	for i := 0; i < len(tests); i++ {
+		result := schelpers.TestResult{}
```
Comment (estroz) [inline internal/scorecard/plugins/plugin_runner.go]: Why not construct the `TestResult` within bracketes?

File: internal/scorecard/plugins/plugin_runner.go [hunk 5]
```diff
@@ -470,5 +481,40 @@ func runTests(csv *olmapiv1alpha1.ClusterServiceVersion, pluginType PluginType,
...
+			tests = append(tests[:i], tests[i+1:]...)
+			i--
+		}
+	}
+	return tests
+}
+
+// testResultToScorecardTestResult is a helper function for converting from the TestResult type
+// to the ScorecardTestResult type
+func testResultToScorecardTestResult(tr schelpers.TestResult) scapiv1alpha2.ScorecardTestResult {
```
Comment (camilamacedo86) [inline internal/scorecard/plugins/plugin_runner.go]:
```suggestion
func newScorecardTestResultFromTestResult(tr schelpers.TestResult) scapiv1alpha2.ScorecardTestResult {
```
nit ^

File: internal/scorecard/plugins/plugin_runner.go (hunk 1 above)
Comment (jmccormick2001) [reply internal/scorecard/plugins/plugin_runner.go]: I think I can fix that, but I'd like to move it to a separate PR if that's ok...in that PR I'd fix this, probably try to remove the helper package altogether, and restructure or combine the Run()/List() into a single chunk of code.

File: internal/scorecard/plugins/labels_test.go (hunk 3 above)
Comment (jmccormick2001) [reply internal/scorecard/plugins/labels_test.go]: I'll give that a try in the next refactor PR, good suggestion.

File: internal/scorecard/plugins/plugin_runner.go (hunk 5 above)
Comment (jmccormick2001) [reply internal/scorecard/plugins/plugin_runner.go]: good suggestion, I want to do another refactor PR around this area and will consider this one in that PR.  thx.

**Pull request merged**